commit 08257610302159e08fd4f5d33787807374ea63c7
Author: Sibi Sankar <quic_sibis@quicinc.com>
Date:   Tue Jun 30 13:49:38 2020 +0530

    arm64: dts: qcom: sc7180: Drop the unused non-MSA SID
    
    Having a non-MSA (Modem Self-Authentication) SID bypassed breaks modem
    sandboxing i.e if a transaction were to originate from it, the hardware
    memory protections units (XPUs) would fail to flag them (any transaction
    originating from modem are historically termed as an MSA transaction).
    Drop the unused non-MSA modem SID on SC7180 SoCs and cheza so that SMMU
    continues to block them.
    
    Tested-by: Douglas Anderson <dianders@chromium.org>
    Reviewed-by: Sai Prakash Ranjan <saiprakash.ranjan@codeaurora.org>
    Reviewed-by: Douglas Anderson <dianders@chromium.org>
    Fixes: bec71ba243e95 ("arm64: dts: qcom: sc7180: Update Q6V5 MSS node")
    Fixes: 68aee4af5f620 ("arm64: dts: qcom: sdm845-cheza: Add iommus property")
    Cc: stable@vger.kernel.org
    Reported-by: Sai Prakash Ranjan <saiprakash.ranjan@codeaurora.org>
    Signed-off-by: Sibi Sankar <sibis@codeaurora.org>
    Link: https://lore.kernel.org/r/20200630081938.8131-1-sibis@codeaurora.org
    Signed-off-by: Bjorn Andersson <bjorn.andersson@linaro.org>

diff --git a/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi b/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
index 70466cc4b405..64fc1bfd66fa 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-cheza.dtsi
@@ -634,7 +634,7 @@ &mdss_mdp {
 };
 
 &mss_pil {
-	iommus = <&apps_smmu 0x780 0x1>,
+	iommus = <&apps_smmu 0x781 0x0>,
 		 <&apps_smmu 0x724 0x3>;
 };